storage/store.go

```diff
@@ -1269,10 +1269,61 @@ func (s *Store) ReplicaCount() int {
 // ExecuteCmd fetches a range based on the header's replica, assembles
 // method, args & reply into a Raft Cmd struct and executes the
 // command using the fetched range.
-func (s *Store) ExecuteCmd(ctx context.Context, args proto.Request) (proto.Response, error) {
+func (s *Store) ExecuteCmd(ctx context.Context, args proto.Request) (reply proto.Response, _ error) {
 	ctx = s.Context(ctx)
 	trace := tracer.FromCtx(ctx)
-	// If the request has a zero timestamp, initialize to this node's clock.
+	if trace == nil {
+		trace = s.ctx.Tracer.NewTrace(args.Header())
+		ctx = tracer.ToCtx(ctx, trace)
+		defer trace.Finalize()
+	}
+	if batch, isBatch := args.(*proto.BatchRequest); isBatch {
```

Comment:
no TODO needed, it's provisional code. It was clear that I would have to write it since I didn't want to get into the Store. I was planning for most of this to go with your PR in. Are you following #2155 at all? Part of this is reflected here by updating everything from the batch except for the key ranges.